### PR TITLE
Add SDL Controller Support

### PR DESCRIPTION
**Pull request by jjbliss**
As requested in issue #56 , here is SDL controller support.
To use, add a command line argument, like "-joy1 SNES" and it will use the first SDL controller.
This works in a similar way to the ps2 implementation.

This might be improved by switching over to SDL events to maintain the controller state, as I've seen it miss very fast button presses.  That issue was with a somewhat laggy bluetooth controller, so I'm not sure how much was just the bluetooth problems though.

### CODE REVIEW

**Comment by jjbliss** [comment]
I've been testing this on MacOS, so it's possible there will be problems on other platforms.  I will go ahead and try it on Linux now.

**Comment by mist64** [comment]
Great work! Just minor comments:
* Can you please be careful with the coding style. Tabs like everywhere else, spaces before and after "{" and "}", space after "if", "i++" instead of "++i". If in doubt, copy the coding style in the rest of the project.
* We don't need "-joy1 none", since this would be the default without specifying "-joy1".
* We need to be consistent about whether we name it "controller" or "joystick". In a Commodore world, it's a joystick, and that's what the KERNAL calls it. I think we should stick with it, so the C symbols, filenames and the help text should say joystick.

File: joystick.c [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+		}
+	}
+	writing = false;
+	return true;
+}
+
+void joystick_step()
+{
+	if (joy1_mode == NONE) {
+		return;
```
Comment (greg-king5) [inline joystick.c]: That code doesn't let us use _only_ `-joy2`!  (We might want the keyboard to be JOY1.)

File: joystick.c [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+	}
+	if (!writing) { //if we are not already writing, check latch to
+		//see if we need to start
+		handle_latch(joystick_latch, joystick_clock);
+	}
+
+	//if we have started writing controller data and the latch has dropped,
+	// we need to start the next bit
+	if (writing & !joystick_latch) {
```
Comment (greg-king5) [inline joystick.c]: The `writing` variable is tested immediately above here; it doesn't need to be tested again.  Therefore, the line would be better as
```c
else if (!joystick_latch) {
```

**Comment by jjbliss** [comment]
Thanks greg, I'll put in those changes.  A few more too.
I've put together a small test program to see the controller inputs.  There are a few problems I'm seeing, so I'll try to fix those.
https://github.com/jjbliss/x16-controller-test

File: joystick.c [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+static uint16_t joystick2_state = 0;
+static uint8_t clock_count = 0;
+
+
+bool joystick_latch, joystick_clock;
+bool joystick1_data, joystick2_data;
+
+bool joystick_init()
+{
+	int joystick1_number = 0;
```
[the remark this answers is not in the document]
Comment (jjbliss) [reply joystick.c]: I've updated this.

File: joystick.c (hunk 1 above)
Comment (jjbliss) [reply joystick.c]: I've done some updates to make sure that joy2 can work at the same time as the keyboard now.  It turns out I wasn't even properly checking joy2_mode.

File: joystick.c (hunk 2 above)
Comment (jjbliss) [reply joystick.c]: I have updated this.

**Pull request merged**